torch_audiomentations/augmentations/impulse_response.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+import random
+import torch
+from ..core.transforms_interface import BasicTransform
+from ..utils.convolution import convolve
+from ..utils.file import find_audio_files, load_audio
+
+
+class ApplyImpulseResponse(BasicTransform):
+    """
+    Convolves an input signal with an impulse response.
+    """
+
+    def __init__(self, ir_path, convolve_mode="full", p=0.5):
+        super(ApplyImpulseResponse, self).__init__(p)
+        self.ir_path = find_audio_files(ir_path)
+        self.convolve_mode = convolve_mode
+
+    def randomize_parameters(self, samples, sample_rate):
+        super(ApplyImpulseResponse, self).randomize_parameters(samples, sample_rate)
+        if self.parameters["should_apply"]:
+            self.parameters["ir_file_path"] = random.choice(self.ir_path)
+
+    def forward(self, samples, sample_rate):
+        super(ApplyImpulseResponse, self).forward(samples, sample_rate)
```

Comment:
Is this really what we want though? 
This should still be `apply`, which will be called by the `BasicTransform`'s `forward` not? So we avoid calling this line every-time?

Reply:
Agreed

Reply:
I guess we can do that too. Though I think to stick with the torch way of subclassing `nn.Module` is to override `forward`.

Reply:
We'll still be compatible with the `nn.Module`, since the superclass implements forward and calls the subclass' apply :)

Reply:
I just noticed that sanity checks for ensuring the input signal is non-empty and if it should apply the augmentation will be eliminated for _all_ subclasses of `BasicTransform`.